org.eclipse.scanning.points/src/org/eclipse/scanning/points/CompoundSpgIterator.java

```diff
@@ -78,65 +76,19 @@ public CompoundSpgIterator(CompoundGenerator gen) throws GeneratorException {
         pyIterator = iterator;
 	}
 
-	private IPosition createFirstPosition() throws GeneratorException {
-
+	private IPosition createFirstPosition(){
 	    IPosition pos = new MapPosition();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "pos" which hides the field declared at line 49. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)